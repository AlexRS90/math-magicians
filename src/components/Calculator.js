import React, { useState } from 'react';
import calculate from '../logic/calculate';


const initial = {
    total: '0',
    next: '',
    operation: '',
};

const Calculator = () => {
  const [obj, newNumbers ] = useState(initial);

  const handleClick = (obj2, btnValue) => {
    const newObj = calculate(obj2, btnValue);
    if (!newObj.next && !newObj.total) {
        newObj.next = '0';
    }
    newNumbers(newObj);
  };

    return (
      <div className="calculator-container">
        <p className="display-result">
          {obj.total}
          {obj.operation}
          {obj.next}
        </p>
        <button onClick={() => handleClick(obj, 'AC')} type="button">AC</button>
        <button onClick={() => handleClick(obj, '+/-')} type="button">+/-</button>
        <button onClick={() => handleClick(obj, '%')} type="button">%</button>
        <button onClick={() => handleClick(obj, 'รท')} type="button" className="operations">&divide;</button>
        <button onClick={() => handleClick(obj, '7')} type="button" className="numbers">7</button>
        <button onClick={() => handleClick(obj, '8')} type="button" className="numbers">8</button>
        <button onClick={() => handleClick(obj, '9')} type="button" className="numbers">9</button>
        <button onClick={() => handleClick(obj, 'x')} type="button" className="operations">X</button>
        <button onClick={() => handleClick(obj, '4')} type="button" className="numbers">4</button>
        <button onClick={() => handleClick(obj, '5')} type="button" className="numbers">5</button>
        <button onClick={() => handleClick(obj, '6')} type="button" className="numbers">6</button>
        <button onClick={() => handleClick(obj, '-')} type="button" className="operations">-</button>
        <button onClick={() => handleClick(obj, '1')} type="button" className="numbers">1</button>
        <button onClick={() => handleClick(obj, '2')} type="button" className="numbers">2</button>
        <button onClick={() => handleClick(obj, '3')} type="button" className="numbers">3</button>
        <button onClick={() => handleClick(obj, '+')} type="button" className="operations">+</button>
        <button onClick={() => handleClick(obj, '0')} type="button" className="numbers zero">0</button>
        <button onClick={() => handleClick(obj, '.')} type="button" className="numbers">.</button>
        <button onClick={() => handleClick(obj, '=')} type="button" className="operations">=</button>
      </div>
    );
}

export default Calculator;
