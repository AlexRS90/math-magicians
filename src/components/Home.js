import React from 'react';

export default function Home() {
  return (
    <div className="container">
      <h2 className="home-title">Welcome to our page</h2>
      <p className="home-introduction">
        October 14th is celebrated as World Maths Day.
        Let us know some interesting and amazing facts about Mathematics.
      </p>
      <p className="home-facts">
        <span className="bullets">1. </span>
        Zero ( 0 ) is the only number which can not be represented by Roman numerals.
        <br />
        <span className="bullets">2. </span>
        What comes after a million, billion and trillion? A quadrillion, quintillion,
        sextillion, septillion, octillion, nonillion, decillion and undecillion
        <br />
        <span className="bullets">3. </span>
        Plus (+) and Minus (-) sign symbols were used as early as 1489 A.D
        <br />
        <span className="bullets">4. </span>
        2 and 5 are the only primes that end in 2 or 5
        <br />
        <span className="bullets">5. </span>
        An icosagon is a shape with 20 sides
        <br />
        <span className="bullets">6. </span>
        Among all shapes with the same perimeter a circle has the largest area.
        <br />
        <span className="bullets">7. </span>
        Abacus is considered the origin of the calculator
        <br />
        <span className="bullets">8. </span>
        40 when written &quot;forty&quot; is the only number with letters in alphabetical order,
        while &quot;one&quot; is the only one with letters in reverse order
        <br />
        <span className="bullets">9. </span>
        &quot;FOUR&quot; is the only number in the English language that is spelt with the same
        number of letters as the number itself
        <br />
        <span className="bullets">10. </span>
        From 0 to 1,000, the letter &quot;A&quot; only appears in 1,000 (&quot;one thousand&quot;)
        <br />
      </p>
        <p className="home-last">
          Words such as formula, equation and calculation sounds boring for
          those who hate Maths as a subject, whereas it is fun for those who
          have keen interest towards solving equations/problems.
        </p>
    </div>
  );
}
