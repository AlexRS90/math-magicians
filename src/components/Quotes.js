import React from 'react';
import { Link } from 'react-router-dom';

export default function Quotes() {
  return (
    <div className="container">
      <h1>PÃ¡gina de Quotes</h1>
      <p>
        <Link to="/edulazaro">Proyectos de Edu</Link>
      </p>
    </div>
  );
}