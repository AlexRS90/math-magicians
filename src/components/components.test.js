import React from "react";
import renderer from 'react-test-renderer';
import Home from "./Home";
import Quotes from './Quotes';
import Calculator from './Calculator';

describe('Rendering all pages', () => {
  it('Render Home page', () => {
    const nav = renderer
      .create(<Home />)
      .toJSON();
    expect(nav).toMatchSnapshot();
  });

  it('Render Quote page', () => {
    const quote = renderer
      .create(<Quotes />)
      .toJSON();
    expect(quote).toMatchSnapshot();
  });

  it ('Render Calculator', () => {
    const calc = renderer
      .create(<Calculator />)
      .toJSON();
    expect(calc).toMatchSnapshot();
  });
});