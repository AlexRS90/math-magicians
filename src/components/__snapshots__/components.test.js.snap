// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering all pages Render Calculator 1`] = `
<div
  className="calc-wrapper"
>
  <p
    className="calc-text"
  >
    Let's do some Math!
  </p>
  <div
    className="calculator-container"
  >
    <p
      className="display-result"
    >
      0
      
      
    </p>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operations"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="numbers"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="numbers"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="numbers"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operations"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
    <button
      className="numbers"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="numbers"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="numbers"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operations"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="numbers"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="numbers"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="numbers"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operations"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="numbers zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="numbers"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operations"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Rendering all pages Render Home page 1`] = `
<div
  className="container"
>
  <h2
    className="home-title"
  >
    Welcome to our page
  </h2>
  <p
    className="home-introduction"
  >
    October 14th is celebrated as World Maths Day. Let us know some interesting and amazing facts about Mathematics.
  </p>
  <p
    className="home-facts"
  >
    <span
      className="bullets"
    >
      1. 
    </span>
    Zero ( 0 ) is the only number which can not be represented by Roman numerals.
    <br />
    <span
      className="bullets"
    >
      2. 
    </span>
    What comes after a million, billion and trillion? A quadrillion, quintillion, sextillion, septillion, octillion, nonillion, decillion and undecillion
    <br />
    <span
      className="bullets"
    >
      3. 
    </span>
    Plus (+) and Minus (-) sign symbols were used as early as 1489 A.D
    <br />
    <span
      className="bullets"
    >
      4. 
    </span>
    2 and 5 are the only primes that end in 2 or 5
    <br />
    <span
      className="bullets"
    >
      5. 
    </span>
    An icosagon is a shape with 20 sides
    <br />
    <span
      className="bullets"
    >
      6. 
    </span>
    Among all shapes with the same perimeter a circle has the largest area.
    <br />
    <span
      className="bullets"
    >
      7. 
    </span>
    Abacus is considered the origin of the calculator
    <br />
    <span
      className="bullets"
    >
      8. 
    </span>
    40 when written "forty" is the only number with letters in alphabetical order, while "one" is the only one with letters in reverse order
    <br />
    <span
      className="bullets"
    >
      9. 
    </span>
    "FOUR" is the only number in the English language that is spelt with the same number of letters as the number itself
    <br />
    <span
      className="bullets"
    >
      10. 
    </span>
    From 0 to 1,000, the letter "A" only appears in 1,000 ("one thousand")
    <br />
  </p>
  <p
    className="home-last"
  >
    Words such as formula, equation and calculation sounds boring for those who hate Maths as a subject, whereas it is fun for those who have keen interest towards solving equations/problems.
  </p>
</div>
`;

exports[`Rendering all pages Render Quote page 1`] = `
<div
  className="quote"
>
  <img
    alt="Math quote"
    className="quotes-img"
    src="51P7XOzA4bL.jpeg"
  />
</div>
`;
